src/Spec.ts

```diff
@@ -503,6 +507,129 @@ export default class Spec {
     }
   }
 
+  private async buildMultipage() {
+    this.log('Creating multipage build...');
+
+    const doc = this.doc.cloneNode(true) as Document;
+
+    const links = new Map<string, string>();
+    const linksForJson: { [key: string]: string[] } = {};
+    const registerLinks = (page: string, node: Element) => {
+      const all = [];
+      if (node.id) {
+        links.set(`#${node.id}`, `${page}#${node.id}`);
+        all.push(node.id);
+      }
+      Array.from(node.querySelectorAll('[id]')).forEach(n => {
+        links.set(`#${n.id}`, `${page}#${n.id}`);
+        if (!n.id.startsWith('_ref')) {
+          all.push(n.id);
+        }
+      });
+      linksForJson[page] = all;
+    };
+
+    const pages = new Map<string, Element>();
+    pages.set('index', doc.createElement('div'));
+    let hasIntro = false;
+    Array.from(doc.querySelector('#spec-container')!.children).forEach(c => {
+      if (!hasIntro) {
+        registerLinks('index.html', c);
+        pages.get('index')!.appendChild(c);
```

Comment:
Couldn't you just stuff the `index` nodes into an array here and only later create the actual container for all of them? That way you could avoid the `// hack` below.